### DISCUSSION TITLE
Django's |date:"..." template filter does not work with TemplateHTMLRenderer

### DISCUSSION THREAD

**Discussion by DanielSwain** [encode/django-rest-framework#8129] [Potential Issue]
I have a viewset that uses TemplateHTMLRenderer as the first `renderer_class`.  The form is returned fine when a date is rendered in an unformatted fashion:
![image](https://user-images.githubusercontent.com/7819909/129374130-771be115-99ca-4d62-bbeb-62c2f4bce5aa.png)

However, when I include a `|date:"..."` filter after `date_created`, then nothing is rendered`:
```
<div class="field">
	<label>Date Created</label>
	{{ date_created|date:"l, M j, Y @ g:i:s A" }}
</div>
```

![image](https://user-images.githubusercontent.com/7819909/129374499-97f39821-4575-42a4-9afd-0c9f86b27f62.png)

All Django built-in template tags work fine (e.g., `{% if %}`, `{% for %}`), but the `|date:"..."` filter does not work for any valid date format string.  One other odd thing is that if I put in an invalid date filtering string (`{{ date_created|date:"xxxxxxxxxxx" }}`), then the invalid formatting string is rendered:

![image](https://user-images.githubusercontent.com/7819909/129375726-c67d5366-3180-4b20-80ee-304d0d5767f1.png)

Is this a bug?

**Comment by DanielSwain** [comment]
It just occurred to me that the date is not actually being treated as a `datetime` object but, rather, it has been converted to a string.  Not sure if this would be considered a bug.

**Reply by mymedia2** [reply]
Perhaps because the template works on serialized data which serializer_class has converted.
